documentation/modules/ROOT/pages/configuration/signalling.adoc

```diff
@@ -564,3 +565,88 @@ endif::community[]
 * {link-prefix}:{link-oracle-connector}#oracle-incremental-snapshots[Oracle connector incremental snapshots]
 * {link-prefix}:{link-postgresql-connector}#postgresql-incremental-snapshots[PostgreSQL connector incremental snapshots]
 * {link-prefix}:{link-sqlserver-connector}#sqlserver-incremental-snapshots[SQL Server connector incremental snapshots]
+
+// Type: procedure
+// ModuleID: debezium-signaling-defining-a-custom-action
+// Title: Defining a custom signal action
+[id="debezium-signaling-custom-action"]
+=== Defining a custom action
+
+Custom actions enable you to extend the {prodname} signaling framework to trigger actions that are not available in the default implementation.
+You can use a custom action with multiple connectors.
+
+To define a custom signal action, you must define the following interface:
+
+
+[source,java,indent=0]
+----
+@FunctionalInterface
+public interface SignalAction<P extends Partition> {
+
+    /**
+     * @param signalPayload the content of the signal
+     * @return true if the signal was processed
+     */
+    boolean arrived(SignalPayload<P> signalPayload) throws InterruptedException;
+}
+
+----
+
+The `io.debezium.pipeline.signal.actions.SignalAction` exposes a single method with one parameter that is the message payloads sent through the signaling channel.
+
+After you define a custom action, you can provide custom actions through the following `io.debezium.pipeline.signal.actions.SignalActionProvider`SPI interface.
```

Comment:
```suggestion
After you define a custom action, you can provide custom actions through the following `io.debezium.pipeline.signal.actions.SignalActionProvider`SPI interface.
```

I have one question about this. I think that we need to differentiate more between _defining a custom action_ and _providing custom actions_ .

So, in the preceding step, you add code to implement a custom action channel (not sure whether `channel` is the appopriate term here). 
After you have done that, you must specify the actual actions that you want to trigger, yes?  Can we change the word `provide` to ``specify`, or does `provide` have special meaning in this context? Maybe something like the following:
```suggestion
After you define a custom signaling action, you can use the following SPI interface to  specify the actions that you want to trigger:  `io.debezium.pipeline.signal.actions.SignalActionProvider`SPI interface.
```

Reply:
Is not about triggering is actually `provide` to be available and usable later sending a signal.